onnxruntime/core/providers/cuda/math/einsum.cc

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "einsum.h"
+
+namespace onnxruntime {
+
+namespace cuda {
+
+ONNX_OPERATOR_KERNEL_EX(
+    Einsum,
+    kOnnxDomain,
+    12,
+    kCudaExecutionProvider,
+    KernelDefBuilder().TypeConstraint("T",
+                                      std::vector<MLDataType>{
+                                          DataTypeImpl::GetTensorType<float>(),
```

Comment:
CUDA MatMul only supportes float and double for now. Hence, Einsum can only support these for now.

Reply:
CUDA MatMul also support MLFloat16. Could we add that as well?

Reply:
Do you have a model with immediate requirement for float16 ? If not, I'd suggest waiting to add support. To purely support float16, I would have to make sure all auxiliary ops support it and I would have to support it in Diagonal.

Reply:
There is no immediate requirement. It is fine to be not in this pull request.
Please add it to back log. Wser would prefer float16 if GPU has tensor core, and having float16 implementation will help them.

Reply:
sure, I ll add it for the backlog for now. Will tackle it soon.